### PR TITLE
Course certificate

### PR DESCRIPTION
**Pull request by elearningforce-mg**
Added a lot of changes to Completion Set, Course Session tests and added new tests for Certificate Template

### CODE REVIEW

File: infrastructure/course-helper.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Constants } from './constants/constants';
 import { CourseCatalogType, CourseSessionInfo, PackageType } from './models';
 import { Helper } from './helpers/helper'
 import { NightWatchBrowser as Browser } from 'nightwatch';
+import { Moment } from 'moment';
 import moment = require("moment");
```
Comment (elearningforce-db) [inline infrastructure/course-helper.ts]: "import moment" must be extra here. Should be removed.

File: infrastructure/helpers/helper.ts [hunk 2]
```diff
@@ -173,32 +172,40 @@ export class Helper extends BrowserHelper {
         return this;
     }
 
-    public toggleCheckbox(needToSelect: boolean, id: string): Browser {
-        this.browser.element('css selector', `[id^='${id}']`, (response) => {
-            this.browser.elementIdSelected(response.value.ELEMENT, (result) => {
-                if ((needToSelect == true && result.value == false) || (needToSelect == false && result.value == true)) {
-                    this.browser.execute(`document.querySelector('[id^=${id}]').click();`);
-                }
+    public toggleCheckbox(needToSelect: boolean, id: string, positionToggle?: number): Browser {
```
Comment (elearningforce-ang) [inline infrastructure/helpers/helper.ts]: Please refactor the method to use it only for old kendo UI

File: infrastructure/learning-module-helper.ts [hunk 3]
```diff
@@ -116,8 +107,7 @@ export class LearningModule {
     }
 
     public dragAndDropItem(value: string): LearningModule {
```
Comment (elearningforce-ang) [inline infrastructure/learning-module-helper.ts]: Please make the method private

File: infrastructure/learning-module-helper.ts [hunk 4]
```diff
@@ -417,43 +386,38 @@ export class LearningModule {
     }
 
     public findLearningModule(text: string): LearningModule {
```
Comment (elearningforce-ang) [inline infrastructure/learning-module-helper.ts]: Maybe the common methods can be used here?

File: infrastructure/learning-module-helper.ts [hunk 5]
```diff
@@ -417,43 +386,38 @@ export class LearningModule {
...
-        this._browser
-            .useXpath()
-            .waitForElementPresent(`//tr[td[contains(text(),"${text}")]]`, Constants.Timeouts.Large)
+        this._browser.useXpath().waitForElementPresent(`//tr[td[contains(text(),"${text}")]]`, Constants.Timeouts.Large);
         return this;
     }
 
     public selectLearningModule(text: string): LearningModule {
```
Comment (elearningforce-ang) [inline infrastructure/learning-module-helper.ts]: Maybe the common methods can be used here?

File: infrastructure/learning-module-helper.ts [hunk 6]
```diff
@@ -417,43 +386,38 @@ export class LearningModule {
...
-        this._browser
-            .useXpath().click(`//tr[td[contains(text(),"${text}")]]//a[@class="actions-link calloutCallLink"]`)
+        this._browser.useXpath().click(`//tr[td[contains(text(),"${text}")]]//a[@class="actions-link calloutCallLink"]`);
         return this;
     }
 
     public checkLMInList(title: string): LearningModule {
```
Comment (elearningforce-ang) [inline infrastructure/learning-module-helper.ts]: Maybe the common methods can be used here?

File: infrastructure/learning-module-helper.ts [hunk 7]
```diff
@@ -417,43 +386,38 @@ export class LearningModule {
...
-        this._browser
-            .useXpath().waitForElementVisible('//div[contains(@id, "learningPathGrid")]', Constants.Timeouts.Large, true)
-            .waitForElementVisible(`//tr[td[contains(text(),"${title}")]]`, Constants.Timeouts.Medium)
+        this._browser.useXpath()
+            .waitForElementVisible('//div[contains(@id, "learningPathGrid")]', Constants.Timeouts.Large, true)
+            .assert.elementPresent(`//tr[td[contains(text(),"${title}")]]`);
         return this;
     }
 
     public checkLMNotInList(title: string): LearningModule {
```
Comment (elearningforce-ang) [inline infrastructure/learning-module-helper.ts]: Maybe the common methods can be used here?

**Pull request merged**